sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/ExtendedSdkLongGauge.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.incubator.metrics.ExtendedLongGauge;
+import io.opentelemetry.api.incubator.metrics.ExtendedLongGaugeBuilder;
+import io.opentelemetry.sdk.metrics.internal.descriptor.Advice;
+import io.opentelemetry.sdk.metrics.internal.descriptor.InstrumentDescriptor;
+import io.opentelemetry.sdk.metrics.internal.state.WriteableMetricStorage;
+import java.util.List;
+
+final class ExtendedSdkLongGauge extends SdkLongGauge implements ExtendedLongGauge {
+
+  private ExtendedSdkLongGauge(
+      InstrumentDescriptor descriptor, SdkMeter sdkMeter, WriteableMetricStorage storage) {
+    super(descriptor, sdkMeter, storage);
+  }
+
+  @Override
+  public boolean isEnabled() {
+    return sdkMeter.isMeterEnabled() && storage.isEnabled();
+  }
+
+  static final class ExtendedSdkLongGaugeBuilder extends SdkLongGaugeBuilder
+      implements ExtendedLongGaugeBuilder {
+
+    ExtendedSdkLongGaugeBuilder(
+        SdkMeter sdkMeter,
+        String name,
+        String description,
+        String unit,
+        Advice.AdviceBuilder adviceBuilder) {
+      super(sdkMeter, name, description, unit, adviceBuilder);
+    }
+
+    @Override
+    public ExtendedSdkLongGauge build() {
+      return builder.buildSynchronousInstrument(ExtendedSdkLongGauge::new);
+    }
+
```

Comment:
just curious, why no `ofDoubles()` here?

Reply:
There's a pattern in which each instrument type has a default or preferred type (i.e. we think most counters are longs, and most gauges are doubles). The builders allow you to convert from the default type to the other type, but not the other way around (i.e. longs to doubles, but not back from doubles to longs).

From a UX perspective, it results in something like this:
```
meter.counterBuilder("counter").build(); // long counter
meter.counterBuilder("counter").ofDoubles().build(); // double counter
meter.counterBuilder("counter").ofDoubles().ofLongs().build(); // <-- not possible since DoubleCounterBuilder does n't have ofLongs();
```

I think the reason is that it never makes sense to do `ofDoubles().ofLongs()` when you can just omit that altogether to get the same result.